contracts/MarginPool.sol

```diff
@@ -130,4 +159,40 @@ contract MarginPool {
             require(transferToUser(_asset[i], _user[i], _amount[i]), "MarginPool: Transfer to user failed");
         }
     }
+
+    /**
+     * @notice function to collect excess balance
+     * @dev can only be called by farmer address
+     * @param _asset asset address
+     * @param _receiver receiver address
+     * @param _amount amount to harvest
+     */
+    function farm(
+        address _asset,
+        address _receiver,
+        uint256 _amount
```

Comment:
why even take an amount? Just pay out the difference between stored and external balance?